WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCouponViewModel.swift

```diff
@@ -221,16 +235,55 @@ final class AddEditCouponViewModel: ObservableObject {
                usedBy: [])
     }
 
-    private enum EditingOption {
+    func validateCouponLocally(_ coupon: Coupon) -> CouponError? {
+        if coupon.code.isEmpty {
+            return .couponCodeEmpty
+        }
+
+        return nil
+    }
+
+    enum EditingOption {
         case creation
         case editing
     }
+
+    enum CouponError: Error, Equatable {
+        case couponCodeEmpty
+        case ciao
```

Comment:
Could you kindly add a comment explaining this error?

Reply:
I forgot to remove it, this is just an error that I introduced for testing that the `Equatable` conformance was working as expected. Removed 👍